### ISSUE TITLE
Feature Request: Support CloudFront Public Keys

### ISSUE THREAD

**Issue by bflad** [terraform-providers/terraform-provider-aws#4096]
Recently the AWS SDK for Go has added support for managing CloudFront public keys as first announced in: https://aws.amazon.com/about-aws/whats-new/2017/12/introducing-field-level-encryption-on-amazon-cloudfront/

Please vote on this issue by adding a :+1: reaction to the original issue to help prioritize interest. If you're interested in implementing this feature request, please comment below.

Prerequisite: aws-sdk-go v.1.13.26 (already merged)

### Terraform Version

terraform 0.10+

### New Resource(s)

- `aws_cloudfront_public_key`

### Terraform Configuration Files

```hcl
# Example implementation, may change when developed
resource "aws_cloudfront_public_key" "example" {
  comment = "" # optional
  encoded_key = "" # required
  name = "" # required
}
```

### Expected Behavior

Resource creates, updates, and deletes CloudFront public keys.

### Actual Behavior

New feature.

### References

* https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/field-level-encryption.html
* https://docs.aws.amazon.com/sdk-for-go/api/service/cloudfront/#CloudFront.CreatePublicKey

**Comment by saravanan30erd** [comment]
@bflad I ll start this resource.

**Comment by bflad** [comment]
This has been released in [version 1.36.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1360-september-13-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
